Makefile

```diff
@@ -0,0 +1,149 @@
+TRIVYADAPTERVERSION=v0.19.0
+
+# version of chartmuseum for pulling the source code
+CHARTMUSEUM_SRC_TAG=v0.13.1
+
+# version of chartmuseum
+CHARTMUSEUMVERSION=$(CHARTMUSEUM_SRC_TAG)-redis
+
+# version prepare
+# for docker image tag
+VERSIONTAG=dev-arm
+BASEIMAGETAG=dev-arm
+BUILD_BASE=true
+PUSHBASEIMAGE=false
+BASEIMAGENAMESPACE=goharbor
+# #input true/false only
+PULL_BASE_FROM_DOCKERHUB=true
+
+# for harbor package name
+PKGVERSIONTAG=dev-arm
+
+PREPARE_VERSION_NAME=versions
+
+# dependency binaries
+CHARTURL=https://storage.googleapis.com/harbor-builds/bin/chartmuseum/release-${CHARTMUSEUMVERSION}/chartm
+NOTARYURL=https://storage.googleapis.com/harbor-builds/bin/notary/release-${NOTARYVERSION}/binary-bundle.tgz
+REGISTRYURL=https://storage.googleapis.com/harbor-builds/bin/registry/release-${REGISTRYVERSION}/registry
+TRIVY_DOWNLOAD_URL=https://github.com/aquasecurity/trivy/releases/download/$(TRIVYVERSION)/trivy_$(TRIVYVERSION:v%=%)_Linux-ARM64.tar.gz
+TRIVY_ADAPTER_DOWNLOAD_URL=https://github.com/aquasecurity/harbor-scanner-trivy/releases/download/$(TRIVYADAPTERVERSION)/harbor-scanner-trivy_$(TRIVYADAPTERVERSION:v%=%)_Linux_x86_64.tar.gz
+
+
+SEDCMD=$(shell which sed)
+SEDCMDI=$(SEDCMD) -i
+ifeq ($(shell uname),Darwin)
+    SEDCMDI=$(shell which gsed) -i
+endif
+
+# dockerhub user
+REGISTRYUSER=
+REGISTRYPASSWORD=
+
+_update_makefile:
+	@echo "update goharbor makefile"
+	@$(SEDCMDI) 's/--rm/--rm --env CGO_ENABLED=0 --env GOOS=linux --env GOARCH=arm64/g' $(HARBOR_MAKEFILE_PATH);
+	@$(SEDCMDI) 's/$$(DOCKERBUILD)/docker buildx build --platform linux\/arm64 --progress plain --output=type=docker/g' $(HARBOR_MAKEFILE_PATH)
+
+_update_make_photon_makefile:
+	@echo "update goharbor photon makefile"
+	@$(SEDCMDI) 's/$(DOCKERCMD) build/$(DOCKERCMD) buildx build --platform linux\/arm64 --progress plain --output=type=docker/' $(HARBOR_PHOTON_MAKEFILE_PATH)
+
+_update_chartserver:
+	@echo "update goharbor chartserver compile.sh"
+	@$(SEDCMDI) 's/go build -a/GOOS=linux GOARCH=arm64 CGO_ENABLED=0 go build -a/g' $(HARBOR_PHOTON_CHARTSERVER_COMPILE_PATH)
+
+_update_registry:
+	@echo "update goharbor Dockerfile.binary"
+	@$(SEDCMDI) 's/CGO_ENABLED=0/GOOS=linux GOARCH=arm64 CGO_ENABLED=0/g' $(HARBOR_PHOTON_REGISTRY_DOCKERFILE_PATH)
+
+_update_trivy-adapter:
+	@echo "update goharbor trivy-adapter Dockerfile.binary"
+	@$(SEDCMDI) 's/CGO_ENABLED=0/GOARCH=arm64 CGO_ENABLED=0/g' $(HARBOR_PHOTON_TRIVY-ADAPTER_DOCKERFILE_PATH)
+
+_update_portal:
+	@echo "update goharbor portal Dockerfile"
+	@$(SEDCMDI) 's/node:15.4.0/--platform=$${BUILDPLATFORM:-linux\/amd64} node:15.4.0/g' $(HARBOR_PHOTON_PORTAL_DOCKERFILE_PATH);
+	@$(SEDCMDI) "s/'node_modules\/@angular\/cli\/bin\/ng'/.\/node_modules\/@angular\/cli\/bin\/ng/g" $(HARBOR_PHOTON_PORTAL_DOCKERFILE_PATH)
+
+_update_notary:
+	@echo "update goharbor notary binary.Dockerfile"
+	@$(SEDCMDI) '8 a ENV CGO_ENABLED 0 \nENV GOOS linux \nENV GOARCH arm64' $(HARBOR_PHOTON_NOTARY_DOCKERFILE_PATH)
+
+_update_pushimage:
```

Comment:
we should use goharbor for all of harbor images, but different tags

Reply:
Yeah, you are right.  I have pushed the local test replacement content up here, and the relevant content will be deleted in the next commit